aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -276,26 +276,29 @@ MPSDataType getMPSScalarType(ScalarType scalar_type) {
   return ss.str();
 }
 
-std::string getTensorsStringKey(const TensorList& tensors, bool short_dtype) {
-  std::string str;
-  // The key format per tensor would look like ":Float32[1,1,1,10]:"
-  for (const Tensor& tensor : tensors) {
-    str += ":";
-    if (tensor.defined()) {
-      str += getMPSTypeString(tensor.scalar_type(), short_dtype) + "[";
-      // if tensor is a scalar
-      if (tensor.dim() == 0) {
-        str += "Scalar";
-      } else {
-        const NSString* ns_shape_key = [[getMPSShape(tensor) valueForKey:@"description"] componentsJoinedByString:@","];
-        str += std::string(ns_shape_key.UTF8String);
-      }
-      str += "]";
-    } else {
-      str += "Undefined";
-    }
-  }
-  return str;
+std::string getTensorsStringKey(const TensorList& tensors, bool short_dtype, bool exclude_shape) {
+    std::string str;
+    // The key format per tensor would look like ":Float32[1,1,1,10]:"
+    for (const Tensor& tensor: tensors) {
+      str += ":";
+      if (tensor.defined()) {
+        str += getMPSTypeString(tensor.scalar_type(), short_dtype) + "[";
+        // if tensor is a scalar
+        if (tensor.dim() == 0) {
+          str += "Scalar";
+        } else {
+          if (exclude_shape) {
```

Comment:
Why would we ever want to exclude shape from the TensorKey?

Reply:
MPS supports unranked placeholders for which we don't need to know the shape to compile the graph - for cases where the graph compilation is very high due to shapes changing very frequently, we can make these ops unranked directly and exclude the shape from the key

Reply:
@DenisVieriu97 Do you mind making this change as a separate PR? Are unranked placeholders supported for all ops or just matmuls? What are performance implications of this one?

Reply:
@malfet, I think these are 2 simple changes which can go together. The one is the change for unranked placeholders and its benefit now.

Reply:
@kulinseth the risk here is that if it causes some regressions it had to be rolled back together